services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/configuration/Configuration.kt

```diff
@@ -63,6 +68,12 @@ class Configuration {
         return CableReportServiceAllu(webClient, alluProps)
     }
 
+    @Bean
+    fun fileScanClient(webClientBuilder: WebClient.Builder): WebClient =
```

Comment:
Won't this interfere with other clients using WebClient?

Reply:
Unless you have a better approach in mind?

Reply:
>According to my understanding, this will be injected to fields named fileScanClient.

I'm under the impression that the type is the important part here. So the value would be injected into all autowired fields with the type WebClient. FileScanService is currently the only place where a WebClient is autowired since the Allu client builds its own instance.

Anyway, it might be better to construct the WebClient where it's needed.
```
class FileScanService(
    webClientBuilder: WebClient.Builder,
    @Value("\${haitaton.clamav.baseUrl}") clamAvUrl: String,
) {
    private val fileScanClient: WebClient =
        webClientBuilder.baseUrl(clamAvUrl).build().also {
            logger.info { "Initialized file scan client with base-url: $clamAvUrl" }
        }
```

Reply:
Fixed as suggested.